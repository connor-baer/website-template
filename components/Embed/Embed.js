import React from 'react';
import styled from '@emotion/styled';
import { css } from '@emotion/core';

const baseStyles = ({ theme, width, height }) => css`
  width: 100%;
  height: 0;
  padding-top: ${((height / width) * 100).toFixed(2)}%;
  background-color: ${theme.colors.offBlack};
  border-radius: ${theme.borderRadius.giga};
  overflow: hidden;
  margin: 2rem 0;
  position: relative;
`;

const fullWidthStyles = ({ fullWidth }) =>
  fullWidth &&
  css`
    grid-column: 1 / 13 !important;
  `;

const Wrapper = styled('div')(baseStyles, fullWidthStyles);

const iframeStyles = () => css`
  max-width: 100%;
  position: absolute;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: 1;
`;

const Iframe = styled('iframe')(iframeStyles);

export default function Embed({ src, title, ...rest }) {
  return (
    <Wrapper {...rest}>
      <Iframe
        src={src}
        title={title}
        allowFullScreen={true}
        frameBorder="0"
        referrerPolicy="no-referrer"
        // eslint-disable-next-line max-len
        sandbox="allow-forms allow-presentation allow-scripts allow-popups allow-popups-to-escape-sandbox"
      />
    </Wrapper>
  );
}
