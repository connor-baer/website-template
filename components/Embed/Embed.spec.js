import React from 'react';

import Embed from './Embed';

describe('Embed', () => {
  function renderEmbed(fn, props = {}) {
    return fn(<Embed {...props} />);
  }

  it('should render with default styles', () => {
    const src = 'https://example.com';
    const title = 'Example website';
    const { container } = renderEmbed(render, { src, title });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should meet basic accessibility guidelines', async () => {
    const src = 'https://example.com';
    const title = 'Example website';
    const html = renderEmbed(renderToHtml, { src, title });
    const result = await axe(html);
    expect(result).toHaveNoViolations();
  });
});
