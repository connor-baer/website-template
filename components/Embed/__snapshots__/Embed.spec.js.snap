// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Embed should render with default styles 1`] = `
.circuit-2 {
  width: 100%;
  height: 0;
  padding-top: NaN%;
  background-color: #15191d;
  border-radius: 6px;
  overflow: hidden;
  margin: 2rem 0;
  position: relative;
}

.circuit-0 {
  max-width: 100%;
  position: absolute;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: 1;
}

<div
  class="circuit-2 circuit-3"
>
  <iframe
    allowfullscreen=""
    class="circuit-0 circuit-1"
    frameborder="0"
    referrerpolicy="no-referrer"
    sandbox="allow-forms allow-presentation allow-scripts allow-popups allow-popups-to-escape-sandbox"
    src="https://example.com"
    title="Example website"
  />
</div>
`;
